modules/form/src/components/field-scopes.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Hint } from "@wso2is/react-components";
+import React, { ReactElement } from "react";
+import { Field as FinalFormField } from "react-final-form";
+import { ScopeFieldAdapter } from "./adapters";
+import { FormFieldPropsInterface } from "./field";
+import { FieldInputTypes, FormFieldMessage } from "../models";
+import { getValidation } from "../utils";
+
+export interface FieldInputPropsInterface extends FormFieldPropsInterface {
+    /**
+     * Type of the input field.
+     */
+    inputType: "default"
```

Comment:
Shall we use an enum instead of defining the list here?

Reply:
this cannot be done as a string needs to be passed to a function and the enum will have to be defined in a way that makes it redundant